src/NServiceBus.Metrics/MetricsFeature.cs

```diff
@@ -1,74 +1,146 @@
+                queueLengthTracker.AttachMetricsContext(metricContext);
+
+                return metricContext;
+            };
+
+            context.RegisterStartupTask(builder => new MetricsReporting(buildContext, metricsOptions.ReportDefintions.ToArray(), builder));
+        }
     }
 
-    static void ConfigureMetrics(FeatureConfigurationContext context, MetricsContext metricsContext)
+    static DefaultMetricsContext BuildMetricContext(string endpointName, Probe[] probes)
     {
-        var builders = context.Settings.Get<MetricsRegistry>().Builders;
+        var context = new DefaultMetricsContext($"{endpointName}");
 
-        foreach (var builder in builders)
+        foreach (var probe in probes)
         {
-            builder.Define(metricsContext);
-            builder.WireUp(context);
+            if (probe.Type == MeasurementValueType.Count)
+            {
+                var counter = context.Counter(probe.Name, string.Empty, default(MetricTags));
+
+                probe.MeasurementTaken += v => counter.Increment(v);
+            }
+            else if (probe.Type == MeasurementValueType.Time)
+            {
+                var timer = context.Timer(probe.Name, string.Empty, SamplingType.LongTerm, TimeUnit.Seconds, TimeUnit.Milliseconds, default(MetricTags));
+
+                probe.MeasurementTaken += v => timer.Record(v, TimeUnit.Milliseconds);
+            }
         }
+
+        return context;
     }
 
     class MetricsReporting : FeatureStartupTask
     {
-        public MetricsReporting(MetricsContext metricsContext, MetricsOptions metricsOptions, IBuilder builder)
+        public MetricsReporting(Func<DefaultMetricsContext> buildContext, MetricsOptions.ReportDefintion[] reportDefinitions, IBuilder builder)
         {
-            this.metricsOptions = metricsOptions;
+            this.buildContext = buildContext;
+            this.reportDefinitions = reportDefinitions;
             this.builder = builder;
-            metricsConfig = new MetricsConfig(metricsContext);
+
+            stopping = new CancellationTokenSource();
         }
 
         protected override Task OnStart(IMessageSession session)
         {
-            metricsOptions.SetUpReports(metricsConfig, builder);
+            foreach (var reportDefinition in reportDefinitions)
+            {
+                var context = buildContext();
+
+                var report = reportDefinition.Builder(builder);
+
+                ReportInIntervals(report, reportDefinition.Interval, context).IgnoreContinuation();
+            }
+
             return Task.FromResult(0);
         }
 
+        async Task ReportInIntervals(MetricsReport report, TimeSpan interval, MetricsContext context)
+        {
+            var reportWindowStart = DateTime.UtcNow;
+
+            while (!stopping.Token.IsCancellationRequested)
+            {
+                try
+                {
+                    await Task.Delay(interval).ConfigureAwait(false);
```

Comment:
can a comment be added here as to why the cancellation token is not passed to `.Delay`